src/engine/INamedContent.ts

```diff
@@ -1,4 +1,4 @@
 export interface INamedContent {
-  name: string;
-  hasValidName: boolean;
+  name: string | null;
+  hasValidName?: boolean;
```

Comment:
The ParsedHierarchy class that implements INamedContent never redeclare `hasValidName` it (or even use it for that matter) so this was the easy way. I'm not familiar enough with C# to know why it allows the interface to be only partially implemented 🤡

Reply:
Alright, I looked at the code in upstream; the two types are different but since we don't currently have namespaces in the project, I guess that works for now. Maybe we should consider putting the runtime and the compiler into different namespaces in the future!